kusto/conn.go

```diff
@@ -54,12 +57,15 @@ func newConn(endpoint string, auth Authorization, client *http.Client) (*conn, e
 		return nil, errors.ES(errors.OpServConn, errors.KClientArgs, "could not parse the endpoint(%s): %s", endpoint, err).SetNoRetry()
 	}
 
-	c := &conn{
-		auth:        auth.Authorizer,
-		endMgmt:     &url.URL{Scheme: "https", Host: u.Host, Path: "/v1/rest/mgmt"},
-		endQuery:    &url.URL{Scheme: "https", Host: u.Host, Path: "/v2/rest/query"},
-		streamQuery: &url.URL{Scheme: "https", Host: u.Host, Path: "/v1/rest/ingest/"},
-		client:      client,
+	c.endMgmt = &url.URL{Scheme: "https", Host: u.Host, Path: "/v1/rest/mgmt"}
+	c.endQuery = &url.URL{Scheme: "https", Host: u.Host, Path: "/v2/rest/query"}
+	c.streamQuery = &url.URL{Scheme: "https", Host: u.Host, Path: "/v1/rest/ingest/"}
+	c.client = client
+
+	if auth.tokenProvider != nil {
+		c.tokenProvider = *auth.tokenProvider
```

Comment:
There may be a question we have to consider, should this be an address. That is, in the client struct should this be *tokenProvider or tokenProvider